### DISCUSSION TITLE
Go Micro most desired features?

### DISCUSSION THREAD

**Discussion by asim** [micro/go-micro#2709] [General]
I'm slowly coming back to my open source projects, Go Micro being the most popular. I'm interested to hear what people's most requested features are and if there are any ideas for how to take this project to the next phase of usage.

**Comment by janu-cambrelen** [comment]
Great to hear, @asim! I've been working with Rust a lot more lately. I'd hesitate to ask for a Rust version of `go-micro` here, but that might require a name change! All jokes aside, your work and this repo inspired me to build something similar in Rust, which I hope to open source at some point. Thanks for your awesome work and good luck with the next phase of development!

As for feature requests, support for gRPC Server Reflection would be fantastic. Judging by [past discussions](https://github.com/go-micro/go-micro/issues/1181) and my own look into it, a desirable solution isn't as straightforward as one might think; still, I think it's worth mentioning here.

All the best!

**Reply by asim** [reply]
Thanks for that and glad to see it inspired you to build something yourself. We actually built a java version alongside Go in the early days. This was done by Sixt the car rental company. It turns out to be a huge undertaking and I think things like Dapr have shown that an alternative approach of baking everything into a server and having client libraries is a better approach. This is where [micro](https://github.com/micro/micro) was going. So if you're looking for that sort of thing take a look over there.

**Comment by outyua** [comment]
so, what's the plan or roadmap on v5, will the new license limit the usage in production? thx

**Reply by asim** [reply]
Well the first point, v5 will require a paid licence subscription to be used in production. Anywhere from $10 to 10000 depending on the size of the org. The second point is that you can continue to use v1-v4 without charge. Those releases are largely fine for everyday use and have tons of plugins and usage. in regards to roadmap, we're going to do that based on paid usage. Those who pay get some sort of say into it. Otherwise we'll just look to start by stabilising the project and make sure bug fixes and releases occur promptly. Then move into bigger things if it makes sense.

**Comment by asim** [comment]
Just a note on my previous comment. I've had some time to think about. I don't think paid licensing is the right answer. So we'll stick with Apache 2 for now. I think the longer term goal is somehow establishing standards in the industry for these building blocks. Go Micro and Micro respectively can play a role in that. We just need to focus on solving problems for the people who are using them right now.
